src/test/fuzz/node_eviction.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <net.h>
+#include <optional.h>
+#include <protocol.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <optional>
+#include <vector>
+
+FUZZ_TARGET(node_eviction)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    std::vector<NodeEvictionCandidate> eviction_candidates;
+    while (fuzzed_data_provider.ConsumeBool()) {
+        eviction_candidates.push_back({
+            fuzzed_data_provider.ConsumeIntegral<NodeId>(),
+            fuzzed_data_provider.ConsumeIntegral<int64_t>(),
+            fuzzed_data_provider.ConsumeIntegral<int64_t>(),
+            fuzzed_data_provider.ConsumeIntegral<int64_t>(),
+            fuzzed_data_provider.ConsumeIntegral<int64_t>(),
+            fuzzed_data_provider.ConsumeBool(),
+            fuzzed_data_provider.ConsumeBool(),
+            fuzzed_data_provider.ConsumeBool(),
+            fuzzed_data_provider.ConsumeIntegral<uint64_t>(),
+            fuzzed_data_provider.ConsumeBool(),
+            fuzzed_data_provider.ConsumeBool(),
```

Comment:
Following this merge I needed to update this fuzzer for #20197 and found the documentation for each data member previously added in `net_tests.cpp` both useful and missing here. Added it in #20197.